src/pydisagg/disaggregate.py

```diff
@@ -164,104 +148,21 @@ def split_dataframe(
             point estimate and standard error for the estimate for each group is given.
     """
     splitting_df = observation_group_membership_df.copy()
-    if use_se is False:
-        def split_row(x):
-            return split_datapoint(
-                x['obs'],
-                population_sizes.loc[x.name]*x[groups_to_split_into],
-                rate_patterns.loc[x['pattern_id']],
-                model=model
-            )
-        result = (
-            splitting_df
-            .set_index('demographic_id')
-            .apply(
-                split_row,
-                axis=1)
-            .reset_index()
-            # .groupby('demographic_id')
-            # .sum()
-        )
-    else:
-        def split_row(x):
-            raw_split_result = split_datapoint(
-                x['obs'],
-                population_sizes.loc[x.name]*x[groups_to_split_into],
-                rate_patterns.loc[x['pattern_id']],
-                model=model,
-                observed_total_se=x['obs_se']
+    if demographic_id_columns is not None:
+        splitting_df['demographic_id']=list(
```

Comment:
Hi @AHsu98, I don't think create `'demgraphic_id'` column is necessary, especially you will delete it at the end of the function. Maybe I am missing something. If you can explain a little here that will be great!

Pandas `set_index` function can take multiple column to create the index, and `reset_index` will flatten out the hierarchical index.

Reply:
Yeah, setting it as demographic id here was kind of a hack, I don't really like it. I set it as a tuple because otherwise I was getting errors around the index name not matching with the population df, I'll make an issue and try to fix it later!